mesonbuild/dependencies/configtool.py

```diff
@@ -115,6 +91,39 @@ def find_config(self, versions: T.List[str], returncode: int = 0) \
 
         return best_match
 
+    def _check_config(self, potential_bin: ExternalProgram, args: T.List[str], versions: T.List[str], returncode: int = 0) \
+            -> T.Tuple[T.Optional[T.List[str]], T.Optional[str]]:
+        if not potential_bin.found():
+            return (None, None)
+        tool = potential_bin.get_command() + args
+        try:
+            p, out = Popen_safe(tool + [self.version_arg])[:2]
+        except (FileNotFoundError, PermissionError):
+            return (None, None)
+        if p.returncode != returncode:
+            if self.skip_version:
+                # maybe the executable is valid even if it doesn't support --version
+                p = Popen_safe(tool + [self.skip_version])[0]
+                if p.returncode != returncode:
+                    return (None, None)
+            else:
+                return (None, None)
+
+        out = self._sanitize_version(out.strip())
+        # Some tools, like pcap-config don't supply a version, but also
+        # don't fail with --version, in that case just assume that there is
+        # only one version and return it.
+        if not out:
+            return (tool, None)
+        if versions:
+            is_found = version_compare_many(out, versions)[0]
+            # This allows returning a found version without a config tool,
+            # which is useful to inform the user that you found version x,
+            # but y was required.
+            if not is_found:
+                tool = None
+        return tool, out
```

Comment:
If this is going to be done, I think it'd be great to take the opportunity to stop calling the version variable `out`, because that is a terrible name.